coral-schema/src/test/resources/testNullabilityExtractUnionUDF-expected.avsc

```diff
@@ -7,12 +7,13 @@
     "type" : [ "null", {
       "type" : "record",
       "name" : "c1",
-      "namespace" : "rel_avro",
+      "namespace" : "namespace_from_first_nested_field_tag_0",
```

Comment:
This sounds to be backward incompatible for DSM, correct?

Reply:
Can we make it more general? It is fine if it needs more code.

Reply:
Can we make this change kick in only when there is ambiguity, and keep the old code path as is?

Reply:
Thanks for the suggestions. I have modified the logic to kick in only when there's ambiguity.
Also, the modifications in the merge utility only impact non-DSM code paths. PTAL